orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+                         "%s cq_progress_handler called for conduitid %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+
+      /* call the progress fn to read the cq and process the message
+        *    for the conduit */
+        count = orte_rml_ofi_progress(conduit);
+        return count;
+}
+
+
+static orte_rml_base_module_t*
+rml_ofi_component_init(int* priority)
+{
+    int ret, fi_version;
+    struct fi_info *hints, *fabric_info;
+    struct fi_cq_attr cq_attr = {0};
+    struct fi_av_attr av_attr = {0};
+    size_t namelen;
+    char   ep_name[FI_NAME_MAX]= {0};
+    uint8_t fabric_id = 0, cur_conduit;
+
+    opal_output_verbose(1,orte_rml_base_framework.framework_output,
+                        "%s - Entering rml_ofi_component_init()",ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
+
+    /*[TODO] Limiting current implementation to APP PROCs.  The PMIX component does not get initialised for
+     *        ORTE_DAEMON_PROC so the OPAL_MODEX_SEND_STRING (opal_pmix.put) fails during initialisation, 
+     *        this needs to be fixed to support DAEMON PROC to use ofi
+    */
+    if (!ORTE_PROC_IS_APP) {
+        *priority = 0;
+        return NULL;
+    }
+
+    if (init_done) {
+        *priority = 2;
+        return &orte_rml_ofi.super;
+    }
+
+    *priority = 2;
+
+
+    /**
+     * Hints to filter providers
+     * See man fi_getinfo for a list of all filters
+     * mode:  Select capabilities MTL is prepared to support.
+     *        In this case, MTL will pass in context into communication calls
+     * ep_type:  reliable datagram operation
+     * caps:     Capabilities required from the provider.
+     *           Tag matching is specified to implement MPI semantics.
+     * msg_order: Guarantee that messages with same tag are ordered.
+     */
+
+    hints = fi_allocinfo();
+    if (!hints) {
+        opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                            "%s:%d: Could not allocate fi_info\n",
+                            __FILE__, __LINE__);
+        return NULL;
+    }
+
+    /**
+     * Refine filter for additional capabilities
+     * endpoint type : Reliable datagram
+     * threading:  Disable locking
+     * control_progress:  enable async progress
+     */
+     hints->mode               = FI_CONTEXT;
+     hints->ep_attr->type      = FI_EP_RDM;      /* Reliable datagram         */
+
+     hints->domain_attr->threading        = FI_THREAD_UNSPEC;
+     hints->domain_attr->control_progress = FI_PROGRESS_AUTO; 
+     hints->domain_attr->data_progress = FI_PROGRESS_AUTO; 
+     hints->domain_attr->av_type          = FI_AV_MAP;
+
+    /**
+     * FI_VERSION provides binary backward and forward compatibility support
+     * Specify the version of OFI is coded to, the provider will select struct
+     * layouts that are compatible with this version.
+     */
+     fi_version = FI_VERSION(1, 0);
```

Comment:
Do you know that this code is compatible with libfabric >= v1.0?  If not, it might be better to just start with the current version of libfabric (v1.3).